product_subscription/models/invoice.py

```diff
@@ -39,18 +39,25 @@ def get_next_start_date(self, subscriber):
 
     @api.multi
     def process_subscription(self, effective_date):
+        # todo should be delegated subscription request
         self.ensure_one()
+        request = self.product_subscription_request
```

Comment:
I know we're not in a controller but using request could be misleading

Reply:
👌

Reply:
👌 good point